### DISCUSSION TITLE
如何快捷的在Swift的Codable协议中，实现继承？

### DISCUSSION THREAD

**Discussion by intsig171** [intsig171/SmartCodable#53] [Q&A]
期望继承的实现可以在封装内部实现，而不是让使用者重写协议方法。

**Comment by intsig171** [comment]
例如提供一个协议用来标记是否继承关系，如果识别到继承，就在内部通过某种方式，调用super.init(from)

**Comment by faimin** [comment]
有位老哥用宏实现了`Codable` 的继承处理，可以借鉴一下。 [CodableWrapper](https://github.com/winddpan/CodableWrapper)
